File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if(s.length!==t.length) return false
    //track all letters
    let lettersS={}
    let lettersT={}
    for (let letter of s){
        if(!lettersS[letter]) lettersS[letter]=1
        else lettersS[letter]+=1
    }
    for (let letter of t){
        if(!lettersT[letter]) lettersT[letter]=1
        else lettersT[letter]+=1
    }
    for (let letter of s){
        if(lettersS[letter]!==lettersT[letter]) return false
    }
    return true
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
# dict to track appearance of each char in first str
# loop through 2nd str, each appearance of char minus 1 in the dict
# if all value in dict is 0 at the end-true
        tracker={}
        for char in s:
            if not char in tracker.keys():
                tracker[char]=0
            tracker[char]+=1
        for char in t:
            if not char in tracker.keys():
                return False
            tracker[char]-=1
        for val in tracker.values():
            if val != 0:
                return False
            
        return True
    
    
        